py_env_create

```diff
@@ -3,34 +3,48 @@
 # py_env_create -- setup the python environment in order to use CTSM python tools
 #
 # Simple bash script to setup the python environment for the user so they can run the CTSM
-# python tools using "conda".
+# python tools using "conda" or "mamba".
 #
+set -e
+trap 'echo "Exiting on Ctrl+C"; exit 130' INT
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "${SCRIPT_DIR}"
+
 dir=${0%/*}
 if [ "$dir" = "$0" ];then
   dir="."
 fi
 
-# Check if conda is in your path
-conda --help >& condahelp.txt
-error=$?
-if [ $error != 0 ]; then
-   echo "conda is NOT in your path for the bash shell add it with modules or whatever is required on your system to get it in your path"
-   echo "on Derecho/capser/etc use -- module load conda"
-   echo "on izumi/CGD systems use -- module unload lang/python; module load lang/anaconda/23.11.0/base"
-   echo "For notes on installing on a user system see: https://docs.conda.io/projects/conda/en/latest/user-guide/install/index.html"
-   echo "Error code was $error"
-   cat condahelp.txt
-   rm condahelp.txt
-   exit -1
+# Try falling back to mamba if conda doesn't work for some reason
+condamamba="conda"
+tmp_env_name="tmp_env_$(date "+%Y%m%d%H%M%S")"
+set +e
+${condamamba} create --name ${tmp_env_name} --yes --quiet 1>/dev/null 2>&1
+result=$?
+set -e
+if [[ ${result} != 0 ]]; then
+   condamamba="mamba"
+   set +e
```

Comment:
OK, I wonder about adding complexity of creating an environment at this stage. There could be disk issues as to why the environment couldn't be created for example. That's the advantage of the dead simple "${condamake} --help" we had before it's something so simple it's got to function. At this first point we just want to make sure that conda is valid. Now, if conda is pointing to something else, using arguments that only conda is likely to know about is also a good thing.

I'm thinking part of what you are doing is removing the complexity of creating a file that has to then be deleted, which is one advantage here...

It is good to know whether you can create a simple environment though. But, maybe this should be a step later?

Reply:
`conda --help` works on my Mac but `conda create` doesn't, which is why I went with the latter for this check.

Reply:
I've made the error message more accurate (b24aba5c9):
```
echo "Neither conda nor mamba seem to be available and working."
echo "Maybe they're not on your path?"
echo "on Derecho/Casper/etc use: module load conda"
...
```